### ISSUE TITLE
How to remove randomness?

### ISSUE THREAD

**Issue by sishue** [philipperemy/keras-tcn#66]
I tried the following things:
`# Set seed value
seed_value = 56
import os
os.environ['PYTHONHASHSEED']=str(seed_value)
# 2. Set `python` built-in pseudo-random generator at a fixed value
import random
random.seed(seed_value)
# 3. Set `numpy` pseudo-random generator at a fixed value
import numpy as np
np.random.seed(seed_value)
from comet_ml import Experiment
# 4. Set `tensorflow` pseudo-random generator at a fixed value
import tensorflow as tf
tf.set_random_seed(seed_value)
# 5. Configure a new global `tensorflow` session
from keras import backend as K
session_conf = tf.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
K.set_session(sess)`

However, I cannot still get the same answer by running several times.

Thank you very much!

Best

**Comment by philipperemy** [comment]
@sishue good point. In tensorflow, it's very difficult to have reproducible results (especially on a GPU).

**Comment by lingdoc** [comment]
Just a heads-up that reproducible results are possible on (NVIDIA) GPUs using the [tensorflow-determinism](https://github.com/NVIDIA/tensorflow-determinism) library. I've been getting reproducible results consistently for CNNs, and I also just tested with `keras-tcn` and got reproducible results, so it is possible.

**Comment by philipperemy** [comment]
@lingdoc yeah you're right. Can you detail the steps to make it reproducible? That would be highly appreciated!

**Comment by philipperemy** [comment]
Is it just adding those lines at the beginning of the script that you run?

```
import tensorflow as tf
from tfdeterminism import patch
patch()
# use tf as normal
```

**Comment by lingdoc** [comment]
yes, after installing the library and setting `SEED` to some value, i.e. `SEED = 54`. I'm running on TF 1.15, btw - haven't tried it on TF 2.0

This should ensure that any sources of randomization, like `shuffle=True` in the Keras `model.fit()` call or the various `split` functions in `sklearn` will be deterministic, provided they are using one of these random seeds. One caveat is that if you change your hardware or model architecture it seems the results will be different, but as long as all the hardware is the same I get the same outcome from running the same model.

**Comment by philipperemy** [comment]
That sounds already very promising! Thank you very much for the feedback. I'm going to add it to the README

**Comment by philipperemy** [comment]
Done! @sishue try the tensorflow-determinism library. It should work!
